databricks/koalas/base.py

```diff
@@ -180,8 +180,13 @@ def __sub__(self, other):
 
     def __truediv__(self, other):
         def truediv(left, right):
-            return F.when(F.lit(right == 0), F.lit(np.inf).__div__(left)).otherwise(
-                left.__truediv__(right)
+            return F.when(
+                F.lit(right == 0) & (F.lit(left != np.inf) & F.lit(left != -np.inf)),
```

Comment:
Why do we need ` & (F.lit(left != np.inf) & F.lit(left != -np.inf))` condition here? Seems if `right` is not `0`, then we will try the next branch.